hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -403,6 +419,41 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		// This new condition is necessary for OCM personnel to report any cloud dangling objects to the user.
+		// The grace period is customizable using an annotation called HCDestroyGracePeriodAnnotation. It's a time.Duration annotation.
+		// This annotation will create a new condition called HostedClusterDestroyed which in conjuntion with CloudResourcesDestroyed
+		// a SRE could determine if there are dangling objects once the HostedCluster is deleted. These cloud dangling objects will remain
+		// in AWS, and SRE will report them to the final user.
+		if r.HCDestroyGracePeriod > 0 {
+			hostedClusterDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.HostedClusterDestroyed))
+			if hostedClusterDestroyedCondition == nil {
+				hostedClusterDestroyedCondition = &metav1.Condition{
+					Type:               string(hyperv1.HostedClusterDestroyed),
+					Status:             metav1.ConditionTrue,
+					Message:            fmt.Sprintf("Grace period set: %v", r.HCDestroyGracePeriod),
+					Reason:             hyperv1.WaitingForGracePeriodReason,
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					ObservedGeneration: hcluster.Generation,
+				}
+
+				meta.SetStatusCondition(&hcluster.Status.Conditions, *hostedClusterDestroyedCondition)
+				if err := r.Client.Status().Update(ctx, hcluster); err != nil {
+					return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
+				}
+			}
+
+			CloudResourcesDestroyed := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+			if CloudResourcesDestroyed.Status == metav1.ConditionFalse && time.Since(hostedClusterDestroyedCondition.LastTransitionTime.Time) <= r.HCDestroyGracePeriod {
+				return ctrl.Result{RequeueAfter: ReportingGracePeriodRequeueDuration}, nil
```

Comment:
You can simply requeue with RequeueAfter: r.HCDestroyGracePeriod -  time.Since(hostedClusterDestroyedCondition.LastTransitionTime.Time)